### DISCUSSION TITLE
Implement AI code generation, Image generation, and Sound generation.

### DISCUSSION THREAD

**Discussion by itzhexen** [twinbasic/twinbasic#1474] [Ideas]
These AI's are good at generating code, Images, and Sounds (even music). The idea it to implement these into the ide or as plugins for the ide.

An example would be to click on a button and in the code editor where there is usually an area to set breakpoints. Well next to it whatever line you are currently one would have a code generation button and you could provide a prompt to an AI which is partially described to generate code for twinbasic and it would give you code back at that line in your code.

Next, say you add a picture box or image box to an application and want a picture but you're not good at art and you want it fast and free. You could also pass a prompt to an AI and get the image directly in your image or picture box and if you like the one it generated it will save it and actually load it into the image or picturebox. if you don't like it maybe have a regenerate button or let it allow you to revise your prompt. The same would go with sounds and music.

**Comment by fafalone** [comment]
It definitely seems like there's a place for "AI" in programming but this seems more along the lines of something that should be handled by community project/3rd party tool rather than a native IDE feature, given how rapidly the tech is advancing. Hopefully at some point tB will have some kind of plugin architecture to give easy access to code info; this would be a great use for it.
